stablehlo/tests/stablehlo_refine_shapes.mlir

```diff
@@ -500,6 +500,15 @@ func.func @refine_dynamic_reshape(%arg0: tensor<4xf32>) -> tensor<*xf32> {
 
 // -----
 
+// Test for #1350
+// CHECK-LABEL: @refine_function_type
+func.func @refine_function_type(%arg0: tensor<3x2xf32>, %arg1: tensor<3x2xf32>) -> (tensor<3x2xf32>, tensor<3x2xf32>) {
+  %0 = stablehlo.add %arg0, %arg1 : tensor<3x2xf32>
+  return %0, %0 : tensor<3x2xf32>, tensor<3x2xf32>
+}
```

Comment:
Does this test actually exercise the new behavior? Let's have a test, similarly to other tests nearby which has static arguments but unranked results, and then refines all the results to be static?

Reply:
Good catch, I coped the output of my test case, not the test itself. Updated so output types all have dynamic shapes.

Reply:
Moved to `update_function_type_multiple_outputs`